File: 144-binary-tree-preorder-traversal/binary-tree-preorder-traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Integer> preorderTraversal(TreeNode root) {
        List<Integer> ll= new ArrayList<>();
        Solve(root,ll);
        return ll;
    }
    public static void Solve(TreeNode root, List<Integer> ll){
        if(root==null){
            return;
        }
        ll.add(root.val);
        Solve(root.left,ll);
        Solve(root.right,ll);
       
    }
}

File: 144-binary-tree-preorder-traversal/binary-tree-preorder-traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<int> preorderTraversal(TreeNode* root) {
        vector<int> ll;
        Solve(root,ll);
        return ll;
    }
    void Solve(TreeNode* root, vector<int>& ll){
        if(root==nullptr){
            return;
        }
        ll.push_back(root->val);
        Solve(root->left,ll);
        Solve(root->right,ll);

    }
};